webknossos-datastore/app/com/scalableminds/webknossos/datastore/services/UploadService.scala

```diff
@@ -125,6 +124,8 @@ class UploadService @Inject()(dataSourceRepository: DataSourceRepository, dataSo
             dataSourceRepository.updateDataSource(
               dataSourceService.dataSourceFromFolder(dataSourceDir, dataSourceId.team))
         case e =>
+          // TODO check whether this is too aggressive
+          PathUtils.deleteDirectoryWithContents(dataSourceDir)
```

Comment:
While testing this PR, I noticed that the Dataset is still reported to wk because of the empty directory that remains after the unsuccessful unzipping. Therefore, I created this method to delete the erroneous `dataSourceDir`, but maybe this too aggressive :thinking:

Reply:
I think this could already solve #4971. I will move the code there and another request to wk to remove the dataset from the db if the unzipping fails. Additionally, we should think about "reserving" the name already for the first request, so similar named datasets fail early and not only in the `finishUpload` request.

Reply:
Cool that you looked into that already :) I think atually deleting is indeed too aggressive, as we may not be 100% certain that this is never wrongly triggered. I’d suggest moving to `.trash` instead (but still double-checking that this really only happens in the error case).
The other ideas sound good to me too, but, as you say, let’s do all of that in a separate PR